code/src/terrat_vcs_github_comment/terrat_vcs_github_comment.ml

```diff
@@ -0,0 +1,298 @@
+module Api = Terrat_vcs_api_github
+module By_scope = Terrat_scope.By_scope
+module Publisher_tools = Terrat_vcs_github_comment_publishers.Publisher_tools
+module Output = Terrat_vcs_github_comment_publishers.Output
+module Scope = Terrat_scope.Scope
+module Tmpl = Terrat_vcs_github_comment_assets.Tmpl
+module Ui = Terrat_vcs_github_comment_assets.Ui
+module Visible_on = Terrat_base_repo_config_v1.Workflow_step.Visible_on
+
+module Sql = struct
+  let read fname =
+    CCOption.get_exn_or
+      fname
+      (CCOption.map Pgsql_io.clean_string (Terrat_files_github_sql.read fname))
+
+  let select_comment_id =
+    Pgsql_io.Typed_sql.(
+      sql
+      //
+      (* comment_id *)
+      Ret.bigint
+      /^ read "select_comment_id.sql"
+      /% Var.uuid "work_manifest"
+      /% Var.text "dir"
+      /% Var.text "workspace")
+
+  let select_comment_elements =
+    Pgsql_io.Typed_sql.(
+      sql
+      //
+      (* work_manifest_id *)
+      Ret.uuid
+      //
+      (* comment.dir *)
+      Ret.text
+      //
+      (* comment.workspace *)
+      Ret.text
+      //
+      (* work_manifest.run_type *)
+      Ret.text
+      //
+      (* steps.ignore_errors *)
+      Ret.boolean
+      //
+      (* steps.payload *)
+      Ret.ud'
+        CCFun.(
+          CCOption.wrap Yojson.Safe.from_string
+          %> CCOption.flat_map
+               (Terrat_api_components_workflow_step_output.Payload.of_yojson %> CCOption.of_result))
+      //
+      (* steps.scope *)
+      Ret.ud'
+        CCFun.(
+          CCOption.wrap Yojson.Safe.from_string
+          %> CCOption.flat_map
+               (Terrat_api_components_workflow_step_output_scope.of_yojson %> CCOption.of_result))
+      //
+      (* steps.success *)
+      Ret.boolean
+      //
+      (* steps.step *)
+      Ret.text
+      /^ read "select_comment_elements.sql"
+      /% Var.bigint "comment_id")
+
+  let upsert_github_work_manifest_comment =
+    Pgsql_io.Typed_sql.(
+      sql
+      //
+      (* comment_id *)
+      Ret.bigint
+      /^ read "upsert_github_work_manifest_comment.sql"
+      /% Var.bigint "comment_id"
+      /% Var.uuid "work_manifest"
+      /% Var.text "dir"
+      /% Var.text "workspace")
+end
+
+module S = struct
+  type t = {
+    account_status : Terrat_vcs_provider2.Account_status.t;
+    client : Api.Client.t;
+    config : Api.Config.t;
+    db : Pgsql_io.t;
+    (* gates : Terrat_gate.t list; *)
```

Comment:
IIRC, does this work with gates btw?  Do we lose historic gate creation?

Reply:
actually, now that you mention, I have no idea why this is commented, let me fix that, wtf.